common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/store/vrang/VraNgBlueprintStore.java

```diff
@@ -241,10 +242,15 @@ private void handleBlueprintImport(final File bpDir, final Map<String, VraNgBlue
 		VraNgBlueprint bp = loadBlueprintFromFilesystem(bpDir);
 		String bpID;
 
+		// Check the blueprint folder name(bpName) with details.json->name(bp.getName())
+		if (!Objects.equals(bp.getName(), bpName)) {
+			throw new IllegalStateException(String.format("Mismatch between blueprint folder name and details.json->name. (%s, %s)", bpName, bp.getName()));
+		}
+
 		// Check if the blueprint exists
 		VraNgBlueprint existingRecord = null;
-		if (bpsOnServerByName.containsKey(bpName)) {
-			existingRecord = bpsOnServerByName.get(bpName);
+		if (bpsOnServerByName.containsKey(bp.getName())) {
```

Comment:
Is the change necessary? After all, we verify bp.getName() equals bpName above.

Reply:
This will ensure the single source of truth for both validation and updating operations. It would be beneficial for future maintenance. If we later decide to remove the exception or do make  other changes on this code, having a consistent source will simplify the things.  WDYT?